src/realm/object-store/impl/realm_coordinator.cpp

```diff
@@ -908,9 +909,12 @@ void RealmCoordinator::run_async_notifiers()
     if (!new_notifiers.empty()) {
         new_notifier_change_info.reserve(new_notifiers.size());
         for (auto& notifier : new_notifiers) {
+            if (notifier->version() == version)
+                continue;
```

Comment:
The additional checks for m_info being null are because of this change; we now no longer call add_required_change_info() if we won't be parsing any transaction logs. I don't think this is a meaningful optimization worth calling out and it's mostly that it made things a bit easier to debug.